### ISSUE TITLE
Campaigns improvements

### ISSUE THREAD

**Issue by muhsin-k** [chatwoot/chatwoot#5845]
# Issues

- [x] https://github.com/chatwoot/chatwoot/issues/4201
- [x] https://github.com/chatwoot/chatwoot/issues/4097
- [x] https://github.com/chatwoot/chatwoot/issues/4026
- [ ] https://github.com/chatwoot/chatwoot/issues/3997
- [x] https://github.com/chatwoot/chatwoot/issues/3624
- [x] https://github.com/chatwoot/chatwoot/issues/3574
- [x] https://github.com/chatwoot/chatwoot/issues/3363
- [x] https://github.com/chatwoot/chatwoot/issues/3192
- [x] https://github.com/chatwoot/chatwoot/issues/3124
- [x] https://github.com/chatwoot/chatwoot/issues/2768
- [x] https://github.com/chatwoot/chatwoot/issues/2739
- [x] https://github.com/chatwoot/chatwoot/issues/2434
- [ ] https://github.com/chatwoot/chatwoot/issues/2431
- [x] https://github.com/chatwoot/chatwoot/issues/2334
- [x] https://github.com/chatwoot/chatwoot/issues/2287
- [x] https://github.com/chatwoot/chatwoot/issues/2282
- [x] #4380
- [x] #4426
- [x] https://github.com/chatwoot/chatwoot/issues/4527
- [x] https://github.com/chatwoot/chatwoot/issues/4531
- [x] https://github.com/chatwoot/chatwoot/issues/4379

**Comment by pauloricardotech** [comment]
O recuso de campanhas pelo WhatsApp seria muito útil

**Comment by rafaelstelles** [comment]
It's possible create campaigns only  some user ? For example, all contacts with custom attribute role equals System Admin.

**Comment by sambitswarup** [comment]
Is there any plan on support rich WhatsApp templates, which have buttons and images or PDF, to be imported and used in the WhatsApp Campaign?

**Comment by muhsin-k** [comment]
> Is there any plan on support rich WhatsApp templates, which have buttons and images or PDF, to be imported and used in the WhatsApp Campaign?

Yes. We are working on it via https://github.com/chatwoot/chatwoot/pull/12117
